main.py

```diff
@@ -175,6 +176,8 @@ def __call__(self, request, response, exception):
       RedirectRoute(r'/_/typeahead/<search_key>', TypeaheadHandler, 'ajax-typeahead', strict_slash=True),
       RedirectRoute(r'/_/webcast/<event_key>/<webcast_number>', WebcastHandler, 'ajax-webcast', strict_slash=True),
       RedirectRoute(r'/_/yt/playlist/videos', YouTubePlaylistHandler, 'ajex-yt-playlist', strict_slash=True),
+      RedirectRoute(r'/<:(frc)?><team_number:[0-9]{1,5}>', ShortTeamHandler, 'short-team-canonical'),
+      RedirectRoute(r'/<event_key:[0-9]{4}[a-z]{1,6}[0-9]?>', ShortEventHandler, 'short-event-detail'),
```

Comment:
Another idea - maybe keep these regexs pretty loose (such as what I wrote above) and only redirect if the Team/Event actually exists. Otherwise show a 404.